lib/filter-stream.js

```diff
@@ -83,79 +79,134 @@ function FilterStream( blockMap, options ) {
 
   constructor: FilterStream,
 
+  get highWaterMark() { return this._writableState.highWaterMark },
+  set highWaterMark( value ) { this._writableState.highWaterMark = value },
+
   /**
-   * Check whether the current position is in a mapped range
+   * Preprocess the `blockMap`'s ranges into byte-ranges
+   * with respect to the `start` offset, and an `offset`
+   * for tracking chunked range reading
+   * @returns {Array<Object>} byteRanges
    * @private
-   * @param {Number} blockAddress
-   * @returns {Boolean}
    */
-  _inRange( blockAddress ) {
-    return this.currentRange &&
-      blockAddress >= this.currentRange.start &&
-      blockAddress <= this.currentRange.end
+  _prepareRanges() {
```

Comment:
Maybe we should find a better name for this function? Something like `getByteRangesFromBlockMap`, as it currently doesn't reflect very well the description you put in JSDoc?